File: back_fil_rouge/app/Models/Cours.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Cours extends Model
{
    use HasFactory;
    protected $fillable = [
        'modules_id',
        'semestres_id',
        'professeurs_id',
        'classes_id',
        'nbr_heure',
    ];
    public function classes()
    {
        return $this->belongsToMany(Classes::class, 'cour_classes', 'cours_id', 'classes_id');
    }

    public function professeurs()
{
    return $this->belongsTo(Professeurs::class, 'professeurs_id');
}
public function module()
{
    return $this->belongsTo(Modules::class, 'modules_id');
}
public function semestre()
{
    return $this->belongsTo(Semestres::class, 'semestres_id');
}
// public function courClasses()
//     {
//         return $this->belongsToMany(CourClasses::class, 'cour_classes', 'cours_id', 'classes_id');
//     }
    public function sessions()
    {
        return $this->hasMany(Sessions::class, 'cours_id'); // Assurez-vous d'ajuster le nom de la clé étrangère si nécessaire
    }

    public function courClasses() {
        return $this->hasMany(CourClasses::class);
    }
}

File: fil_rouge_moble/lib/model/Cours.dart
class Cours {
  final int id;
  final int modulesId;
  final int semestresId;
  final int professeursId;
  final String nbrHeure;
  final String professeurs;
  final String semestre;
  final String module;

  Cours({
    required this.id,
    required this.modulesId,
    required this.semestresId,
    required this.professeursId,
    required this.nbrHeure,
    required this.professeurs,
    required this.semestre,
    required this.module,
  });

  factory Cours.fromJson(Map<String, dynamic> json) {
    return Cours(
      id: json['id'],
      modulesId: json['modules_id'],
      semestresId: json['semestres_id'],
      professeursId: json['professeurs_id'],
      nbrHeure: json['nbr_heure'],
      professeurs: json['professeurs'],
      semestre: json['semestre'],
      module: json['module'],
    );
  }
}
